### PR TITLE
Add travis, fix test cases

### PR DESCRIPTION
**Pull request by Rafiot**

### CODE REVIEW

**Comment by Rafiot** [comment]
(fixing the conflict)

**Comment by Rafiot** [comment]
Ok, you did plenty of changes in the code, the tests are mostly broken now :)

File: setup.py [hunk 1]
```diff
@@ -33,12 +33,5 @@
         ('share/man/man1', ['mat.1', 'mat-gui.1']),
         ('share/nautilus-python/extensions', ['nautilus/nautilus-mat.py'])
     ],
-    cmdclass={
-        'build': build_extra.build_extra,
-        'build_i18n': build_i18n.build_i18n,
-        'build_help': build_help.build_help,
-        'build_icons': build_icons.build_icons,
-        'clean': clean_i18n.clean_i18n,
-    },
```
Comment (jvoisin) [inline setup.py]: Why removing this?

**Comment by jvoisin** [comment]
Yeah, sorry about the conflict :/
My debian maintainer wanted to be able to run the test-suite outside of the source folder, so I added a `--local` and a `--system` flag, to test either the *system-wide* mat version, or the one of the source folder.

File: setup.py (hunk 1 above)
Comment (Rafiot) [inline setup.py]: Because if you use setuptools, and setup.cfg, you don't need it.

**Comment by Rafiot** [comment]
oh, ok, I see. 

The changes I made probably break that. I'll check. 
Also, the tests were failing because exiftool wasn't installed, new commit incoming to fix travis.

File: setup.py (hunk 1 above)
Comment (jvoisin) [inline setup.py]: Perfect

**Comment by jvoisin** [comment]
♥

**Comment by Rafiot** [comment]
So yeah, `--system` fails now, and `--local` will fail if test.py isn't run from the root dir of the repo.
The recommended way is to use the python tests: 
```
python setup.py test
```
Can you ping your packager is this way of running the tests is acceptable? Before I try to make all the path relative and stuff.

**Comment by jvoisin** [comment]
He just wanted to be able to be able to run tests outside the build directory (and I want to still be able to run them on my local version of MAT).

So if I can issue `python setup.py test` to test the local version and something else for the system-wide one (or the other way around), it's ok of me.

Does it make sense to you?

**Comment by Rafiot** [comment]
Sounds good, I'll fix the system-wide test thing :)

**Comment by Rafiot** [comment]
Seems fixed.

**Comment by jvoisin** [comment]
It seems that you factorised the way to decide if it's a local or system-wide test only in clitest, and not in libtest, is this on purpose?

Also, how shall I test local-wide/system-wide ?

**Comment by Rafiot** [comment]
Yes, libtest does't need to 'know' about local/system: it doesn't call the `mat` command.

Sure, feel free to try: you should be able to call test.py -s from anywhere, and the local test is done via `setup.py test` or `test/test.py`

**Comment by jvoisin** [comment]
This is what I got:

```
jvoisin@kaa 21:19 ~/dev/GSoC/mat git pull https://github.com/Rafiot/MAT.git travis
From https://github.com/Rafiot/MAT
 * branch            travis     -> FETCH_HEAD
Updating d054e31..b9e55f6
Fast-forward
 .travis.yml             |  29 +++++++++++++++++++++++++++++
 __init__.py             |   0
 setup.cfg               |   4 ++++
 setup.py                |  13 +++----------
 test/__init__.py        |   2 ++
 test/clitest.py         |  35 ++++++++++++++++++++++-------------
 test/libtest.py         |  10 ++++++----
 test/not_writtable.docx | Bin 0 -> 2300 bytes
 test/test.py            |  23 ++++++++++++++---------
 9 files changed, 80 insertions(+), 36 deletions(-)
 create mode 100644 .travis.yml
 create mode 100644 __init__.py
 create mode 100644 setup.cfg
 create mode 100644 test/not_writtable.docx
 mode change 100644 => 100755 test/test.py
jvoisin@kaa 21:19 ~/dev/GSoC/mat ls
data/    nautilus/  test/       CHANGELOG    dirty é.svg  LICENSE      mat*   mat.desktop  mat-gui.1@       poc.py  README.security  setup.cfg  TESTING
libmat/  po/        351493.png  clean é.svg  __init__.py  MANIFEST.in  mat.1  mat-gui*     mcafee-exif.jpg  README  RELEASE          setup.py*
jvoisin@kaa 21:19 ~/dev/GSoC/mat python test/test.py -s                                                                                           [Rafiot-travis] git:mat
Traceback (most recent call last):
  File "test/test.py", line 78, in <module>
    import clitest
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 28, in <module>
    from libmat import mat
ImportError: No module named libmat
zsh: exit 1     python test/test.py -s
jvoisin@kaa 21:19 ~/dev/GSoC/mat python setup.py test
running test
Searching for gi
Reading https://pypi.python.org/simple/gi/
Best match: gi 1.2
Downloading https://pypi.python.org/packages/source/g/gi/gi-1.2.tar.gz#md5=63bdf817281813f049064133ed584a6e
Processing gi-1.2.tar.gz
Writing /tmp/easy_install-XOL72_/gi-1.2/setup.cfg
Running gi-1.2/setup.py -q bdist_egg --dist-dir /tmp/easy_install-XOL72_/gi-1.2/egg-dist-tmp-IjvTgN
creating /home/jvoisin/dev/GSoC/mat/.eggs/gi-1.2-py2.7.egg
Extracting gi-1.2-py2.7.egg to /home/jvoisin/dev/GSoC/mat/.eggs

Installed /home/jvoisin/dev/GSoC/mat/.eggs/gi-1.2-py2.7.egg
running egg_info
creating MAT.egg-info
writing requirements to MAT.egg-info/requires.txt
writing MAT.egg-info/PKG-INFO
writing top-level names to MAT.egg-info/top_level.txt
writing dependency_links to MAT.egg-info/dependency_links.txt
writing manifest file 'MAT.egg-info/SOURCES.txt'
reading manifest file 'MAT.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'TODO'
warning: no files found matching 'clean.*' under directory 'test'
warning: no files found matching 'dirty.*' under directory 'test'
writing manifest file 'MAT.egg-info/SOURCES.txt'
running build_ext
test_archive_unwritable_content (test.libtest.TestArchiveProcessing) ... ok
test_get_unsupported (test.libtest.TestArchiveProcessing)
Test the get_unsupported feature, used by the GUI ... ok
test_remove_bz2 (test.libtest.TestArchiveProcessing)
Test MAT's ability to process .tar.bz2 ... ok
test_remove_gz (test.libtest.TestArchiveProcessing)
Test MAT on tar.gz files ... ok
test_remove_tar (test.libtest.TestArchiveProcessing)
Test MAT on tar files ... ok
test_empty (test.libtest.TestFileAttributes)
test MAT's behaviour on empty file ... ok
test_not_exist (test.libtest.TestFileAttributes)
test MAT's behaviour on non-existent file ... ok
test_list (test.libtest.TestListlib)
check if get_meta returns metadata ... ok
testlist_list_empty (test.libtest.TestListlib)
check that a listing of a clean file returns an empty dict ... ok
test_remove (test.libtest.TestRemovelib)
make sure that the lib remove all compromizing meta ... ok
test_remove_empty (test.libtest.TestRemovelib)
Test removal with clean files ... ok
test_remove_existing (test.libtest.TestSecureRemove)
test the secure removal of an existing file ... ok
test_remove_fail (test.libtest.TestSecureRemove)
test the secure removal of an non-removable file ... ok
test_clean (test.libtest.TestisCleanlib)
test is_clean on clean files ... ok
test_dirty (test.libtest.TestisCleanlib)
test is_clean on dirty files ... ok
test_empty (test.clitest.TestFileAttributes)
test MAT's behaviour on empty file ... ERROR
test_not_exist (test.clitest.TestFileAttributes)
test MAT's behaviour on non-existent file ... ERROR
test_not_writtable (test.clitest.TestFileAttributes)
test MAT's behaviour on non-writable file ... ERROR
test_list_clean (test.clitest.TestListcli)
check if get_meta returns meta ... ERROR
test_list_dirty (test.clitest.TestListcli)
check if get_meta returns all the expected meta ... ERROR
test_remove (test.clitest.TestRemovecli)
make sure that the cli remove all compromizing meta ... ERROR
test_remove_empty (test.clitest.TestRemovecli)
Test removal with clean files ... ERROR
test_abort_unsupported (test.clitest.TestUnsupported)
test if the cli aborts on unsupported files ... ERROR
test_clean (test.clitest.TestisCleancli)
test is_clean on clean files ... ERROR
test_dirty (test.clitest.TestisCleancli)
test is_clean on dirty files ... ERROR

======================================================================
ERROR: test_empty (test.clitest.TestFileAttributes)
test MAT's behaviour on empty file
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 118, in test_empty
    proc = subprocess.Popen([MAT_PATH, 'test/empty_file'], stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_not_exist (test.clitest.TestFileAttributes)
test MAT's behaviour on non-existent file
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 112, in test_not_exist
    stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_not_writtable (test.clitest.TestFileAttributes)
test MAT's behaviour on non-writable file
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 105, in test_not_writtable
    stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_list_clean (test.clitest.TestListcli)
check if get_meta returns meta
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 60, in test_list_clean
    stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_list_dirty (test.clitest.TestListcli)
check if get_meta returns all the expected meta
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 69, in test_list_dirty
    stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_remove (test.clitest.TestRemovecli)
make sure that the cli remove all compromizing meta
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 39, in test_remove
    subprocess.call([MAT_PATH, '--add2archive', dirty])
  File "/usr/lib/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_remove_empty (test.clitest.TestRemovecli)
Test removal with clean files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 46, in test_remove_empty
    subprocess.call([MAT_PATH, '--add2archive', clean])
  File "/usr/lib/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_abort_unsupported (test.clitest.TestUnsupported)
test if the cli aborts on unsupported files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 133, in test_abort_unsupported
    proc = subprocess.Popen([MAT_PATH, tarpath], stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_clean (test.clitest.TestisCleancli)
test is_clean on clean files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 84, in test_clean
    stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

======================================================================
ERROR: test_dirty (test.clitest.TestisCleancli)
test is_clean on dirty files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jvoisin/dev/GSoC/mat/test/clitest.py", line 92, in test_dirty
    stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

----------------------------------------------------------------------
Ran 25 tests in 131.440s

FAILED (errors=10)
zsh: exit 1     python setup.py test
jvoisin@kaa 21:22 ~/dev/GSoC/mat 
```

Tests are failing, am I doing something wrong?

Also, I don't get why it's downloading `gi`, while I already have it installed on my system.

**Comment by jvoisin** [comment]
I have committed and pushed several of your improvements.

I can't manage to deal with `setuptools`, because it wants to install my package as an `egg`. I have nothing against that, but it makes it really difficult to develop MAT, since I'll have to implement two methods to get the data files : one for when MAT is installed (using eggs), and one for when I'm developing (using regular files).
If you have some ideas about how to do this the proper way, I'm all ears.

Also, regarding the test-suite, I'd like to simplify/complete/refactor/whatever the `setup.py` file before, since it's closely-related.

Thank you for your patience :)

**Comment by Rafiot** [comment]
If you want to use the system wide test, you need to do `python setup.py install` first.

**Comment by Rafiot** [comment]
I'll look at your commits soon-ish.

The main goal of that pull request is to be able to run the tests on travis automatically, I have no strong feelings pro/against `setuptools`, I just want to run the tests via travis :)

**Comment by Rafiot** [comment]
I'm still using setuptools in this commit, because I cannot find how to get distutils to work properly :/

FYI, the way I dev in python is in a virtualenv (pew is the best thing ever), where I can make sure I only have my local dependencies and nothing else.

**Comment by jvoisin** [comment]
The installation process produces failing things here when MAT is installed system-wide:

```
root@kaa 12:02 ~/dev/GSoC/mat python setup.py install                                                                                                  U [master] git:mat
running install
running bdist_egg
running egg_info
writing requirements to MAT.egg-info/requires.txt
writing MAT.egg-info/PKG-INFO
writing top-level names to MAT.egg-info/top_level.txt
writing dependency_links to MAT.egg-info/dependency_links.txt
reading manifest file 'MAT.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'TODO'
warning: no files found matching 'clean.*' under directory 'test'
warning: no files found matching 'dirty.*' under directory 'test'
writing manifest file 'MAT.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/libmat
creating build/bdist.linux-x86_64/egg/libmat/bencode
copying build/lib.linux-x86_64-2.7/libmat/bencode/__init__.py -> build/bdist.linux-x86_64/egg/libmat/bencode
copying build/lib.linux-x86_64-2.7/libmat/bencode/bencode.py -> build/bdist.linux-x86_64/egg/libmat/bencode
copying build/lib.linux-x86_64-2.7/libmat/parser.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/exiftool.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/office.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/strippers.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/audio.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/misc.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/archive.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/images.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/exceptions.py -> build/bdist.linux-x86_64/egg/libmat
creating build/bdist.linux-x86_64/egg/libmat/hachoir_editor
copying build/lib.linux-x86_64-2.7/libmat/hachoir_editor/fieldset.py -> build/bdist.linux-x86_64/egg/libmat/hachoir_editor
copying build/lib.linux-x86_64-2.7/libmat/hachoir_editor/field.py -> build/bdist.linux-x86_64/egg/libmat/hachoir_editor
copying build/lib.linux-x86_64-2.7/libmat/hachoir_editor/__init__.py -> build/bdist.linux-x86_64/egg/libmat/hachoir_editor
copying build/lib.linux-x86_64-2.7/libmat/hachoir_editor/typed_field.py -> build/bdist.linux-x86_64/egg/libmat/hachoir_editor
copying build/lib.linux-x86_64-2.7/libmat/mutagenstripper.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/__init__.py -> build/bdist.linux-x86_64/egg/libmat
copying build/lib.linux-x86_64-2.7/libmat/mat.py -> build/bdist.linux-x86_64/egg/libmat
byte-compiling build/bdist.linux-x86_64/egg/libmat/bencode/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/bencode/bencode.py to bencode.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/parser.py to parser.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/exiftool.py to exiftool.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/office.py to office.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/strippers.py to strippers.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/audio.py to audio.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/misc.py to misc.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/archive.py to archive.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/images.py to images.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/exceptions.py to exceptions.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/hachoir_editor/fieldset.py to fieldset.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/hachoir_editor/field.py to field.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/hachoir_editor/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/hachoir_editor/typed_field.py to typed_field.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/mutagenstripper.py to mutagenstripper.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmat/mat.py to mat.pyc
installing package data to build/bdist.linux-x86_64/egg
running install_data
creating build/bdist.linux-x86_64/egg/share
creating build/bdist.linux-x86_64/egg/share/applications
copying mat.desktop -> build/bdist.linux-x86_64/egg/share/applications
creating build/bdist.linux-x86_64/egg/share/mat
copying data/FORMATS -> build/bdist.linux-x86_64/egg/share/mat
copying data/mat.glade -> build/bdist.linux-x86_64/egg/share/mat
creating build/bdist.linux-x86_64/egg/share/pixmaps
copying data/mat.png -> build/bdist.linux-x86_64/egg/share/pixmaps
creating build/bdist.linux-x86_64/egg/share/doc
creating build/bdist.linux-x86_64/egg/share/doc/mat
copying README -> build/bdist.linux-x86_64/egg/share/doc/mat
copying README.security -> build/bdist.linux-x86_64/egg/share/doc/mat
creating build/bdist.linux-x86_64/egg/share/man
creating build/bdist.linux-x86_64/egg/share/man/man1
copying mat.1 -> build/bdist.linux-x86_64/egg/share/man/man1
copying mat-gui.1 -> build/bdist.linux-x86_64/egg/share/man/man1
creating build/bdist.linux-x86_64/egg/share/nautilus-python
creating build/bdist.linux-x86_64/egg/share/nautilus-python/extensions
copying nautilus/nautilus-mat.py -> build/bdist.linux-x86_64/egg/share/nautilus-python/extensions
creating build/bdist.linux-x86_64/egg/EGG-INFO
installing scripts to build/bdist.linux-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-2.7/mat -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-2.7/mat-gui -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/mat to 755
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/mat-gui to 755
copying MAT.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MAT.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MAT.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MAT.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MAT.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating 'dist/MAT-0.5.3-py2.7.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing MAT-0.5.3-py2.7.egg
Removing /usr/local/lib/python2.7/dist-packages/MAT-0.5.3-py2.7.egg
Copying MAT-0.5.3-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Removing MAT 0.5.4 from easy-install.pth file
Adding MAT 0.5.3 to easy-install.pth file
Installing mat-gui script to /usr/local/bin
Installing mat script to /usr/local/bin

Installed /usr/local/lib/python2.7/dist-packages/MAT-0.5.3-py2.7.egg
Processing dependencies for MAT==0.5.3
Searching for gi
Reading https://pypi.python.org/simple/gi/
Best match: gi 1.2
Downloading https://pypi.python.org/packages/source/g/gi/gi-1.2.tar.gz#md5=63bdf817281813f049064133ed584a6e
Processing gi-1.2.tar.gz
Writing /tmp/easy_install-A_nwJC/gi-1.2/setup.cfg
Running gi-1.2/setup.py -q bdist_egg --dist-dir /tmp/easy_install-A_nwJC/gi-1.2/egg-dist-tmp-hule7y
creating /usr/local/lib/python2.7/dist-packages/gi-1.2-py2.7.egg
Extracting gi-1.2-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Adding gi 1.2 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/gi-1.2-py2.7.egg
Searching for hachoir-parser==1.3.4
Best match: hachoir-parser 1.3.4
Adding hachoir-parser 1.3.4 to easy-install.pth file

Using /usr/lib/python2.7/dist-packages
Searching for hachoir-core==1.3.3
Best match: hachoir-core 1.3.3
hachoir-core 1.3.3 is already the active version in easy-install.pth

Using /usr/lib/python2.7/dist-packages
Searching for pdfrw==0.1
Best match: pdfrw 0.1
pdfrw 0.1 is already the active version in easy-install.pth

Using /usr/lib/python2.7/dist-packages
Searching for mutagen==1.30
Best match: mutagen 1.30
mutagen 1.30 is already the active version in easy-install.pth

Using /usr/lib/python2.7/dist-packages
Searching for requests==2.7.0
Best match: requests 2.7.0
requests 2.7.0 is already the active version in easy-install.pth

Using /usr/lib/python2.7/dist-packages
Finished processing dependencies for MAT==0.5.3
```

When launching MAT
```
jvoisin@kaa 12:02 ~ mat-gui 
Traceback (most recent call last):
  File "/usr/local/bin/mat-gui", line 4, in <module>
    __import__('pkg_resources').run_script('MAT==0.5.3', 'mat-gui')
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 735, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1659, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/MAT-0.5.3-py2.7.egg/EGG-INFO/scripts/mat-gui", line 6, in <module>
    
ImportError: No module named repository
zsh: exit 1     mat-gui
```

**Comment by jvoisin** [comment]
I managed to add automatic Travis support, it's still WIP, but it's almost working :)

**Comment by jvoisin** [comment]
Weehee, it's [done]( https://travis-ci.org/jvoisin/MAT )

**Comment by Rafiot** [comment]
\o/ well done!

**Comment by jvoisin** [comment]
What else do you wanted to achieve with this PR?

**Comment by Rafiot** [comment]
just travis. so as you just did it, you can just close this one :)

**Comment by jvoisin** [comment]
Thank you very much for the idea and for helping me :)

**Pull request closed without merging**